### DISCUSSION TITLE
Configuring Stage-Based Domain Names for server-side-website

### DISCUSSION THREAD

**Discussion by renereed1** [getlift/lift#402] [General]
Hi maintainers,

First, I want to thank you for the excellent work on the serverless-lift plugin. It has significantly streamlined our serverless deployments.

I have a question regarding the configuration of stage-based domain names using the server-side-website construct. Specifically, I'm looking to set up different domains based on the deployment stage. Here's the scenario I'm trying to achieve:

Stage: prod
Domain: example.com

Stage: dev
Domain: dev.example.com

Stage: abc
Doamin: abc.example.com

Here is an example of my current serverless.yml configuration:

service: my-service

provider:
  name: aws
  region: us-west-2
  stage: ${opt:stage, 'dev'}
  runtime: provided.al2

plugins:
  - serverless-lift

custom:
  baseDomain: example.com
  domain:
    'Fn::If':
      - IsProd
      - ${self:custom.baseDomain}
      - ${self:provider.stage}.${self:custom.baseDomain}

constructs:
  website:
    type: server-side-website
    assets:
      '/assets/*': public/assets
    domain: ${self:custom.domain}
    certificate: arn:aws:acm:region:account-id:certificate/certificate-id

resources:
  Conditions:
    IsProd:
      Fn::Equals:
        - ${self:provider.stage}
        - prod

Unfortunately, this configuration results in an invalid format error. Is there a recommended approach to dynamically set domain names based on the stage using the server-side-website construct in serverless-lift?

I appreciate any guidance or examples you can provide. Thanks in advance for your assistance!

Best regards,

**Comment by mnapoli** [comment]
I would recommend to use stage parameters: https://www.serverless.com/framework/docs/guides/parameters#stage-parameters

**Reply by renereed1** [reply]
This is perfect, but available in serverless v4. Is there a recommended approach for v3? The issue I am seeing is 

custom:
baseDomain: example.com
domain:
'Fn::If':
- IsProd
- ${self:custom.baseDomain}
- ${self:provider.stage}.${self:custom.baseDomain}

would work, but Cloudfront is configured before the environment based domain is generated.

Thanks,

**Reply by mnapoli** [reply]
it is available in v3 under the `params` key, it's a shame they removed that from the docs:

```yml
params:
  default:
    domain: ${sls:stage}.preview.myapp.com
  prod:
    domain: myapp.com
  dev:
    domain: preview.myapp.com
```

Then:

```yml
provider:
  environment:
    APP_DOMAIN: ${param:domain}
```

**Reply by renereed1** [reply]
Ah, Works wonderfully. Thank you!
